Sources/UIKit/TextAlignmentConstraint.swift

```diff
@@ -0,0 +1,268 @@
+                    return .Bottom
+                #endif
+            case .unspecified:
+                #if swift(>=3.0)
+                    return .notAnAttribute
+                #else
+                    return .NotAnAttribute
+                #endif
+            }
+        }
+
+        static private let ibInspectableMapping: [String: Attribute] = [
+            "unspecified": .unspecified,
+            "top": .top,
+            "capheight": .capHeight,
+            "xheight": .xHeight,
+            "firstbaseline": .firstBaseline,
+            "lastbaseline": .lastBaseline,
+            "bottom": .bottom
+        ]
+
+        public var description: String {
+            switch self {
+            case .unspecified:
+                return "unspecified"
+            case .top:
+                return "top"
+            case .capHeight:
+                return "cap height"
+            case .xHeight:
+                return "x-height"
+            case .firstBaseline:
+                return "first baseline"
+            case .lastBaseline:
+                return "last baseline"
+            case .bottom:
+                return "bottom"
+            }
+        }
+
+        init(ibInspectableString string: String) {
+            self = Attribute.ibInspectableMapping[string] ?? .unspecified
+        }
+    }
+
+    @IBInspectable public var firstAlignment: String? {
+        didSet {
+            firstItemAttribute = BonMot.TextAlignmentConstraint.Attribute(ibInspectableString: firstAlignment?.normalized ?? "")
+        }
+    }
+
+    @IBInspectable public var secondAlignment: String? {
+        didSet {
+            secondItemAttribute = BonMot.TextAlignmentConstraint.Attribute(ibInspectableString: secondAlignment?.normalized ?? "")
+        }
+    }
+
+    public private(set) var firstItemAttribute: BonMot.TextAlignmentConstraint.Attribute = .unspecified
+    public private(set) var secondItemAttribute: BonMot.TextAlignmentConstraint.Attribute = .unspecified
+    private var item1: AnyObject!
+    private var item2: AnyObject!
+
+    // The class part of these selectors are ignored; it is there simply to satisfy Xcode's selector syntax.
+    #if os(OSX)
+        #if swift(>=3.0)
+        private static let fontSelector = #selector(getter: NSTextField.font)
+        private static let attributedTextSelector = #selector(getter: NSTextField.attributedStringValue)
+        #else
+        private static let fontSelector = Selector("font")
+        private static let attributedTextSelector = Selector("attributedStringValue")
+        #endif
+    #else
+        #if swift(>=3.0)
+        private static let fontSelector = #selector(getter: UITextField.font)
+        private static let attributedTextSelector = #selector(getter: UITextField.attributedText)
+        #else
+        private static let fontSelector = Selector("font")
+        private static let attributedTextSelector = Selector("attributedText")
+        #endif
+    #endif
```

Comment:
This is an interesting one. You should be able to to drop the OS difference by using a typealias and using BONTextField (which will map to NS or UI). Then there's just 1 #if conditional